src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -112,6 +112,10 @@ public bool MapInboundClaims
                 if (!_mapInboundClaims && value && _inboundClaimTypeMap.Count == 0)
                     _inboundClaimTypeMap = new Dictionary<string, string>(DefaultInboundClaimTypeMap);
 
+                // If the inbound claim type mapping is on and being turned off, make sure that the _inboundClaimTypeMap is empty.
+                if (_mapInboundClaims && !value && _inboundClaimTypeMap.Count > 0)
+                    _inboundClaimTypeMap.Clear();
```

Comment:
This is logically what I believe should happen - but is there a reason why we didn't have this before?

Reply:
Quick thought: `_mapInboundClaims` flag should be checked if it's true before map is used. If it's false, the map should never be used. So this should not be needed.

Reply:
Do we have the same behavior in JsonWebTokenHandler?

Reply:
The test tests that the constructor of JwtSecurityTokenHandler sets up the map or not depending if the static flag DefaultMapInboundClaims was set or not; then it also checks if setting MapInboundClaims property initializes the map list.

Both MapInboundClaims and InboundClaimTypeMap are public. In our code we check the flag first before using the map. There are a couple of usage scenarios.

1. MapInboundClaims is false and InboundClaimTypeMap is empty. Set flag to true. This will initialize the map to the default map values, which makes sense since we don't know if the map will be explicitly initialized, so we set it to default. The map can still be overwritten to non-default values.
2. InboundClaimTypeMap is set to custom values. MapInboundClaims is set to true. This will not reset the map to default values because it already as values.
3. InboundClaimTypeMap is set to custom values. MapInboundClaims is set to true, then set to false. With the previous code, the map will not be cleared and when the flag is set to true, custom values will remain. With updated code, the custom values will be cleared, and when the flag is set to true, map will be set to default values. This is the possible breaking change. Probably unlikely but still.

So I would only update the test to reset DefaultMapInboundClaims to true (default) and move it and add a collection attribute to it.

Reply:
These are good points - I appreciate the thoughtful write up and reflection. I've reverted the src code changes and reset the property in the test so that the other tests aren't affected - running this test and the other two previously flaky tests "to failure" for 5 minutes didn't incur any failures whereas previously they would immediately fail.